config/document_types.yml

```diff
@@ -31,7 +31,7 @@ publication: &publication
   pre_release: true
   path_prefix: /government/publications
   lead_image: false
-  attachments: featured_and_inline
+  attachments: featured
```

Comment:
This is a better name.

I think the document type could just say "I have featured" attachments, and the rules in the code could determine what that means.

For example, the document type config doesn't need to configure that it can't "insert" via the inline attachments modal because it has feature attachments.  Does that make sense?

P.S. I assume this is commit is going to be fixed up? "foo" 🤨

Reply:
Ah, that's basically what I wrote in the original card (`featured_attachments: true`). The problem I had is that the insert/inline attachment actions and screens are very different between the news and publications, and both can't be used at the same time. Representing that with a single boolean smells because `featured_attachments: false` actually means turning something else on; representing with two booleans (`featured_attachments` and `inline_file_attachments`) also smells, because this means we can write config with both set to `true`, and would want a validation to stop that. Since we don't really have time to redesign our config to represent this better, I thought an enum/string was the best compromise.

Oops - fixed the commit!

Reply:
Does that help clarify the approach? Some of this is also written on the card.

Reply:
So as far as I am aware documents can have both "inline" and "featured" attachments, because that is how the attachment ends being rendered.

To me, if you said that "publications" don't have inline attachments, what I hear is: "Attachments can't be inserted to the body of the text using the drop-down menu in the markdown editor in the body field".

The fact that for publications, the "inline" attachments can only be selected and inserted via the markdown editor, and not uploaded and inserted doesn't change that they are still rendered inside the body text.

So when I said, that the code not the config determines the behaviour, that's what I meant. That we can say all document types allow inline attachments, but if featured attachments is true, then attachments can't be uploaded via the markdown editor, only inserted. But if featured attachments is false/nil, then attachments can be both uploaded and inserted via the markdown editor.